target/linux/generic/files/drivers/net/phy/ar8327.c

```diff
@@ -1268,6 +1279,35 @@ ar8327_sw_set_igmp_v3(struct switch_dev *dev,
 	return 0;
 }
 
+static int
+ar8327_sw_set_port_vlan_prio(struct switch_dev *dev, const struct switch_attr *attr,
+			     struct switch_val *val)
+{
+	struct ar8xxx_priv *priv = swdev_to_ar8xxx(dev);
+	int port = val->port_vlan;
+
+	if (port >= dev->ports)
+		return -EINVAL;
+	if (port == 0 || port == 6)
+		return -EOPNOTSUPP;
```

Comment:
Since swconfig doesn't enforce the .max attribute, a range check like
<pre>
if (val->value.i < 0 || val->value.i > 7)
  	return -EINVAL;
</pre>
should be done here. Then the masking wouldn't be needed at all anywhere else.

Reply:
done